### PR TITLE
Tmedia 288 card list react optimizations

### PR DESCRIPTION
**Pull request by JackHowa**
## Description
Reduce dom nodes and maintain style. Use images properly sized

## Jira Ticket
- [TMEDIA-288](https://arcpublishing.atlassian.net/secure/RapidBoard.jspa?rapidView=875&modal=detail&selectedIssue=TMEDIA-288&assignee=5e387d6a1b1d910e5dfd7a9b)

## Acceptance Criteria
Optimize for reducing dom nodes without layout regressions

## Test Steps
- Add test steps a reviewer must complete to test this PR

1. Checkout this branch `git checkout TMEDIA-288-card-list-react-optimizations`
2. Use engine theme sdk branch https://github.com/WPMedia/engine-theme-sdk/pull/376
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/card-list-block`
3. Go to http://localhost/pf/homepage/?_website=the-gazette and scroll down to see card list. Compare with corecomponents
Screen Shot 2021-06-25 at 12 20 42
4. Use lighthouse to measure how many dom nodes

### Before
Screen Shot 2021-06-25 at 09 39 14
https://corecomponents.arcpublishing.com/pf/card-list-many-elements/?_website=the-sun

### After
Screen Shot 2021-06-25 at 12 04 09
- create page with just card list and "the-sun" site. and add a query for 50 type:story to replicate the live site 
- see comments for perf improvements https://arcpublishing.atlassian.net/secure/RapidBoard.jspa?rapidView=875&modal=detail&selectedIssue=TMEDIA-288

## Dependencies or Side Effects
- Removed hr elements because it does not reflect "a thematic break between paragraph-level elements" and we want to optimize for reducing dom nodes. I recreated the hr visually element with css; some browsers may experience this differently
- Also removed an unused div that had no styles (only a class name, at least styles from us) associated with it

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [x] Confirmed this PR has unit test files
  - [x] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [ ] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

**Comment by JackHowa** [comment]
nested npmignore is preferred over the unused one higher in the folder structure. not really perf, but cleanup

**Review by matthewroach — DISMISSED** [review]

File: blocks/card-list-block/features/card-list/default.jsx [hunk 1]
```diff
@@ -34,13 +34,13 @@ class CardList extends React.Component {
...
 
-    this.samllImageOptions = {
+    this.smallImageOptions = {
       smallWidth: 105,
       smallHeight: 70,
       mediumWidth: 105,
       mediumHeight: 70,
-      largeWidth: 274,
-      largeHeight: 183,
+      largeWidth: 105,
```
Comment (JackHowa) [inline blocks/card-list-block/features/card-list/default.jsx:42]: removed unused widths due to hard-coded css

File: blocks/card-list-block/features/card-list/default.jsx [hunk 2]
```diff
@@ -34,13 +34,13 @@ class CardList extends React.Component {
       resizerURL: this.siteProperties?.resizerURL,
     };
 
-    this.samllImageOptions = {
+    this.smallImageOptions = {
```
Comment (JackHowa) [inline blocks/card-list-block/features/card-list/default.jsx:37]: minor typo

**Review by matthewroach — APPROVED** [review]
Looks good

**Pull request merged**